### PR TITLE
Filesystem access asynchronization

### PR DESCRIPTION
**Pull request by hobinjk-ptc**

### CODE REVIEW

**Comment by hobinjk-ptc** [comment]
Ohp, more files need asyncification actually

File: libraries/hardwareInterfaces.js [hunk 1]
```diff
@@ -849,8 +849,8 @@ exports.setHardwareInterfaceSettings = function (interfaceName, settings, limitT
  * Complement to setup() which is necessary due to the unique positioning of
  * the setHardwareInterfaceSettings function
  */
-exports.setHardwareInterfaceSettingsImpl = function (setHardwareInterfaceSettings) {
-    setHardwareInterfaceSettingsImpl = setHardwareInterfaceSettings;
+exports.setSetHardwareInterfaceSettingsImpl = function (newSetHardwareInterfaceSettingsImpl) {
```
Comment (dangond-ptc) [inline libraries/hardwareInterfaces.js:852]: Not the biggest fan of `setSet`, but I see what's going on here

**Review by dangond-ptc — APPROVED** [review]
Looks good to me!

File: libraries/hardwareInterfaces.js (hunk 1 above)
Comment (hobinjk-ptc) [reply libraries/hardwareInterfaces.js]: [8258313](https://github.com/ptcrealitylab/vuforia-spatial-edge-server/pull/907/commits/825831351f339b8e9242a8c483b806275d5d4b53) has good news for you

**Review by benptc — APPROVED** [review]
Lots of good changes here! Thanks for the extensive refactoring 🥇

**Comment by hobinjk-ptc** [comment]
Manually merged because GitHub was having a time. After this, any problems with the server can be reported to me for debugging because they're probably my fault

**Pull request closed without merging**